modules/reindex/src/main/java/org/elasticsearch/index/reindex/ReindexRequest.java

```diff
@@ -90,6 +93,9 @@ public ActionRequestValidationException validate() {
         if (destination.timestamp() != null) {
             e = addValidationError("setting timestamp on destination isn't supported. use scripts instead.", e);
         }
+        if (getRemoteInfo() != null && getSearchRequest().source().query() != null) {
+            e = addValidationError("reindex from remote sources should use RemoteInfo's query instead of source's query", e);
```

Comment:
I don't think users will know what `RemoteInfo` is (since it's a java class), perhaps make it look more like the REST API?